homeassistant/components/sensibo/climate.py

```diff
@@ -273,64 +273,45 @@ async def async_set_temperature(self, **kwargs) -> None:
             else:
                 return
 
-        result = await self._async_set_ac_state_property(
+        if await self._async_set_ac_state_property(
             "targetTemperature", int(temperature)
-        )
-        if result:
-            self.coordinator.data[self.unique_id]["target_temp"] = int(temperature)
-            self.async_write_ha_state()
+        ):
+            await self.async_write_result("target_temp", int(temperature))
 
     async def async_set_fan_mode(self, fan_mode: str) -> None:
         """Set new target fan mode."""
-        result = await self._async_set_ac_state_property("fanLevel", fan_mode)
-        if result:
-            self.coordinator.data[self.unique_id]["fan_mode"] = fan_mode
-            self.async_write_ha_state()
+        if await self._async_set_ac_state_property("fanLevel", fan_mode):
+            await self.async_write_result("fan_mode", fan_mode)
 
     async def async_set_hvac_mode(self, hvac_mode: str) -> None:
         """Set new target operation mode."""
         if hvac_mode == HVAC_MODE_OFF:
-            result = await self._async_set_ac_state_property("on", False)
-            if result:
-                self.coordinator.data[self.unique_id]["on"] = False
-                self.async_write_ha_state()
+            if await self._async_set_ac_state_property("on", False):
+                await self.async_write_result("on", False)
             return
 
         # Turn on if not currently on.
         if not self.coordinator.data[self.unique_id]["on"]:
-            result = await self._async_set_ac_state_property("on", True)
-            if result:
+            if await self._async_set_ac_state_property("on", True):
                 self.coordinator.data[self.unique_id]["on"] = True
 
-        result = await self._async_set_ac_state_property(
-            "mode", HA_TO_SENSIBO[hvac_mode]
-        )
-        if result:
-            self.coordinator.data[self.unique_id]["hvac_mode"] = HA_TO_SENSIBO[
-                hvac_mode
-            ]
-            self.async_write_ha_state()
+        if await self._async_set_ac_state_property("mode", HA_TO_SENSIBO[hvac_mode]):
+            await self.async_write_result("hvac_mode", HA_TO_SENSIBO[hvac_mode])
 
     async def async_set_swing_mode(self, swing_mode: str) -> None:
         """Set new target swing operation."""
-        result = await self._async_set_ac_state_property("swing", swing_mode)
-        if result:
-            self.coordinator.data[self.unique_id]["swing_mode"] = swing_mode
-            self.async_write_ha_state()
+        if await self._async_set_ac_state_property("swing", swing_mode):
+            await self.async_write_result("swing_mode", swing_mode)
```

Comment:
You could save a few lines in the entity but moving this into a function and making a mapping 

AC_STATE_PROPERTY_TO_DATA = {
	"swing": "swing_mode",
         "on": "on",
         "fanLevel": "fan_mode"
         "mode": "hvac_mode",
         "targetTemperature": "target_temp"
}

.....

Reply:
Made most sense to put it inside _async_set_ac_state_property directly so it's now cleaned up

Reply:
Made most sense to put it inside _async_set_ac_state_property directly so it's now cleaned up